src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfoCacheService.cs

```diff
@@ -38,40 +48,35 @@ public SymbolTreeInfoCacheService()
             if (metadataId == null)
                 return null;
 
-            var checksum = SymbolTreeInfo.GetMetadataChecksum(solution, reference, cancellationToken);
-
-            // See if the last value produced matches what the caller is asking for.  If so, return that.
-            if (_metadataIdToInfo.TryGetValue(metadataId, out var metadataInfo) &&
-                metadataInfo.SymbolTreeInfo.Checksum == checksum)
-            {
+            // See if the last value produced exactly matches what the caller is asking for.  If so, return that.
+            if (_metadataIdToInfo.TryGetValue(metadataId, out var metadataInfo))
                 return metadataInfo.SymbolTreeInfo;
-            }
 
             // If we didn't have it in our cache, see if we can load it from disk.
-            // Note: pass 'loadOnly' so we only attempt to load from disk, not to actually
-            // try to create the metadata.
-            var info = await SymbolTreeInfo.GetInfoForMetadataReferenceAsync(
-                solution, reference, checksum, loadOnly: true, cancellationToken).ConfigureAwait(false);
-            return info;
+            var info = await SymbolTreeInfo.LoadAnyInfoForMetadataReferenceAsync(solution, reference, cancellationToken).ConfigureAwait(false);
+            if (info is null)
+                return null;
+
+            var referencingProjects = new HashSet<ProjectId>(solution.Projects.Where(p => p.MetadataReferences.Contains(reference)).Select(p => p.Id));
+
+            // attempt to add this item to the map.  But defer to whatever is in the map now if something else beat us to this.
+            return _metadataIdToInfo.GetOrAdd(metadataId, new MetadataInfo(info, referencingProjects)).SymbolTreeInfo;
         }
 
-        public async Task<SymbolTreeInfo?> TryGetSourceSymbolTreeInfoAsync(
+        public async Task<SymbolTreeInfo?> TryGetPotentiallyStaleSourceSymbolTreeInfoAsync(
             Project project, CancellationToken cancellationToken)
         {
-            // See if the last value produced matches what the caller is asking for.  If so, return that.
-            var checksum = await SymbolTreeInfo.GetSourceSymbolsChecksumAsync(project, cancellationToken).ConfigureAwait(false);
-            if (_projectIdToInfo.TryGetValue(project.Id, out var projectInfo) &&
-                projectInfo.Checksum == checksum)
-            {
+            // See if the last value produced exactly matches what the caller is asking for.  If so, return that.
```

Comment:
Same as https://github.com/dotnet/roslyn/pull/63449/files#r953091508

Reply:
can fix in followup PR :)